go/bundle/decoder.go

```diff
@@ -303,134 +304,158 @@ var knownSections = map[string]struct{}{
 }
 
 // https://wicg.github.io/webpackage/draft-yasskin-dispatch-bundled-exchanges.html#load-metadata
-func loadMetadata(bs []byte) (*meta, error) {
+func loadMetadata(bs []byte) (*meta, error, *url.URL) {
```

Comment:
I'd like to keep `error` as the last return arg list.

two ideas:
1. reorder the return arg list, or
2. create a struct which implements error interface, which allows fallbackUrl extraction

Reply:
Created LoadMetadataError struct which embeds an error interface.